react/src/components/App.jsx

```diff
@@ -42,8 +43,9 @@ const App = () => {
             <Route path="/signup" exact component={Signup} />
             <Route path="/studenthome" exact component={StudentHome} />
             <Route path="/teacherhome" exact component={TeacherHome} />
-            <Route path="/userprofile" exact component={UserProfile} />
-            <Route path="/userprofilelang" exact component={UserProfileLang} />
+            <Route path="/userprofile" render={() => <UserProfile userType={userType}/>} />
```

Comment:
Let's have the user profile and reviews pages be components of the user profile layout. And using state render each of the components according to which menu item is selected. That way they aren't their own individual url routes and all under the umbrella of the userprofile page